src/main/java/org/jabref/logic/importer/fetcher/DoiFetcher.java

```diff
@@ -47,18 +51,28 @@ public String getName() {
     @Override
     public Optional<BibEntry> performSearchById(String identifier) throws FetcherException {
         Optional<DOI> doi = DOI.parse(identifier);
+        String agency = "";
         try {
             if (doi.isPresent()) {
-                URL doiURL = new URL(doi.get().getURIAsASCIIString());
+                Optional<BibEntry> fetchedEntry;
+
+                // mEDRA does not return a parsable bibtex string
+                if (getAgency(doi.get()).isPresent() && "medra".equalsIgnoreCase(getAgency(doi.get()).get())) {
+                    fetchedEntry = new Medra().performSearchById(identifier);
```

Comment:
Why is no `doPostCleanup` executed in this case? I would do it here for consitency, too.

Reply:
I would test for the agency and directly return the fetched entry. No else branch necessary --> quick exit and less nesting

Reply:
doPostCleanup is performed inside performSearchById of Medra class. 

Else removed.